runtime/src/main/java/org/corfudb/runtime/LRSiteDiscoveryListener.java

```diff
@@ -0,0 +1,89 @@
+package org.corfudb.runtime;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.collections.CorfuStore;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This listener will notify when a new remote site for specific
+ * log replication model is available for replication.
+ */
+@Slf4j
+public abstract class LRSiteDiscoveryListener {
```

Comment:
Trying to recall why this was needed....is this understanding correct? : essentially this is a temp fix because currently there is no way otherwise for the sink to figure out whats getting replicated?

Reply:
yes, the Sink creates listeners corresponding to each source cluster.  Currently, the is no way for the Routing Queue client to know these clusters.  I will add a TODO in the comment.